### DISCUSSION TITLE
Should we split gnome-shell from the X applications group?

### DISCUSSION THREAD

**Discussion by cakrit** [netdata/netdata#13491] [Ideas]
`gnome-shell` on a default ubuntu installation can take a significant amount of memory. See what happens to the X application group, if we add an additional line/dimension for `gnome-shell`

<img width="485" alt="image" src="https://user-images.githubusercontent.com/43294513/183110903-5a24b494-ead1-49b9-997e-fa0734350a38.png">

So any reason not to change the default config and show it as a separate dimension?

**Comment by cakrit** [comment]
Actually, let me generalize this. I never liked how many things we've thrown into the `X` process group. Am I the only one that often wishes we had more granural groupings? e.g. browsers can be their own category for sure:

```
X: X Xorg xinit lightdm xdm pulseaudio gkrellm xfwm4 xfdesktop xfce* Thunar
X: xfsettingsd xfconfd gnome-* gdm gconf* dconf* xfconf* *gvfs gvfs* slim
X: kdeinit* kdm plasmashell
X: evolution-* firefox chromium opera vivaldi-bin epiphany WebKit*
X: '*systemd --user*' chrome *chrome-sandbox* *google-chrome* *chromium* *firefox*
```

**Comment by ilyam8** [comment]
> Am I the only one that often wishes we had more granural groupings? 

No, not the only one. I agree that more granular groupings are better.

**Comment by MrZammler** [comment]
+1

Even so more than now most applications like gnome-shell run fine under wayland with no X.org running.

**Comment by Ferroin** [comment]
Quick update regarding this:

With the merge of https://github.com/netdata/netdata/pull/13618, the group in question has been renamed to `gui` and expanded slightly to also pick up some popular Wayland tooling.

In my spare time I’ve been working on deaggregating this group, with the current plan being to have a separate group for each DE/WM/Compositor (so GNOME will be it’s own group, KDE will be it’s own group, XFCE will be it’s own group, Sway will be it’s own group, etc), as well as putting the web browsers in their own group, and (probably) having a separate group _just_ covering the X server (independent of the other GUI stuff).

**Reply by Ferroin** [reply]
Preliminary set of changes to implement this here: https://github.com/netdata/netdata/pull/13631
